adoptopenjdk-api-v3-updater/src/main/kotlin/net/adoptopenjdk/api/v3/stats/GithubDownloadStatsCalculator.kt

```diff
@@ -53,23 +54,40 @@ class GithubDownloadStatsCalculator {
     public fun getStats(repos: AdoptRepos): List<GithubDownloadStatsDbEntry> {
         val date: ZonedDateTime = TimeSource.now()
         val stats = repos
-            .repos
-            .values
-            .map { featureRelease ->
-                val total = featureRelease
-                    .releases
-                    .getReleases()
-                    .filter { it.vendor == Vendor.adoptopenjdk }
-                    .sumBy {
-                        it.download_count.toInt()
-                    }
+                .repos
+                .values
+                .map { featureRelease ->
+                    val total = featureRelease
+                            .releases
+                            .getReleases()
+                            .filter { it.vendor == Vendor.adoptopenjdk }
+                            .sumBy {
+                                it.download_count.toInt()
+                            }
 
-                GithubDownloadStatsDbEntry(date,
-                    total.toLong(),
-                    featureRelease.featureVersion
-                )
-            }
-            .toList()
+                    // Tally up jvmImpl download stats
+                    val jvmImplMap: Map<JvmImpl, Long> = JvmImpl.values().map { jvmImpl ->
+                        jvmImpl to
+                            featureRelease
+                            .releases
+                            .getReleases()
+                            .filter { it.vendor == Vendor.adoptopenjdk }
+                            .sumBy {
+                                it.binaries
+                                .filter { it.jvm_impl == jvmImpl }
+                                .sumBy {
+                                    it.download_count.toInt()
```

Comment:
I doubt the toInt and the toLong below are required here

Reply:
Its just how the typing works, sumBy expects an Int to be returned and I made jvm download counts a long to match the overall download count